superset-frontend/src/explore/components/ControlPanelsContainer.tsx

```diff
@@ -161,29 +178,39 @@ class ControlPanelsContainer extends React.Component {
         validationErrors={validationErrors}
         actions={actions}
         formData={provideFormDataToProps ? formData : null}
-        datasource={formData?.datasource}
```

Comment:
Remove this exposed `datasource` prop from #12609  since it's already available in `formData`, plus The typing would not conform with `datasource` for `DatasourceControl`, which uses the whole metadata of a datasource. 

cc @pkdotson @zhaoyongjie